source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -276,8 +276,16 @@ bool Wavm::initialize(const std::string& code, absl::string_view name, bool allo
 
 void Wavm::start(Context* context) {
   auto f = getStartFunction(moduleInstance_);
-  if (!f)
+  if (f) {
+    CALL_WITH_CONTEXT(invokeFunctionChecked(context_, f, {}), context);
```

Comment:
Is there a early return? It's possible that `f` is reassigned and invoked at line 287

Reply:
So, we want to call the generic start function *before* the global variables are initialized.  I am not sure why, but that is what WAVM does and this is WAVM specific code.   The "start" function is special and according to the docs should be called (if available) before any exported functions.

Reply:
Thank you for the explanation!